drivers/kw41zrf/kw41zrf_netdev.c

```diff
@@ -0,0 +1,1283 @@
+            case NETOPT_LAST_ED_LEVEL:
+                kw41zrf_set_power_mode(dev, KW41ZRF_POWER_IDLE);
+#ifdef MODULE_NETOPT
+                DEBUG("[kw41zrf] Wake to get opt %s\n", netopt2str(opt));
+#else
+                DEBUG("[kw41zrf] Wake to get opt %d\n", (int)opt);
+#endif
+                put_to_sleep_when_done = 1;
+                break;
+
+            default:
+                break;
+        }
+    }
+    else {
+        /* Wait for oscillator ready signal if the CPU is coming out of low
+         * power mode */
+        while((RSIM->CONTROL & RSIM_CONTROL_RF_OSC_READY_MASK) == 0) {}
+    }
+
+    int res = -ENOTSUP;
+
+    switch (opt) {
+        case NETOPT_TX_POWER:
+            if (len != sizeof(int16_t)) {
+                res = -EOVERFLOW;
+                break;
+            }
+            *((uint16_t *)value) = kw41zrf_get_txpower(dev);
+            res = len;
+            break;
+
+        case NETOPT_IS_CHANNEL_CLR:
+            if (len != sizeof(netopt_enable_t)) {
+                res = -EOVERFLOW;
+                break;
+            }
+            *((netopt_enable_t *)value) = !(kw41zrf_cca(dev));
+            res = len;
+            break;
+
+        case NETOPT_CCA_THRESHOLD:
+            if (len != sizeof(int8_t)) {
+                res = -EOVERFLOW;
+                break;
+            }
+            *((int8_t *)value) = kw41zrf_get_cca_threshold(dev);
+            res = len;
+            break;
+
+        case NETOPT_CCA_MODE:
+            if (len != sizeof(uint8_t)) {
+                res = -EOVERFLOW;
+                break;
+            }
+            uint8_t mode = kw41zrf_get_cca_mode(dev);
+            switch (mode) {
+                case NETDEV_IEEE802154_CCA_MODE_1:
+                case NETDEV_IEEE802154_CCA_MODE_2:
+                case NETDEV_IEEE802154_CCA_MODE_3:
+                    *((uint8_t *)value) = mode;
+                    res = len;
+                    break;
+                default:
+                    res = -EINVAL;
+                    break;
+            }
+            break;
+
+        case NETOPT_LAST_ED_LEVEL:
+            if (len != sizeof(int8_t)) {
+                res = -EOVERFLOW;
+                break;
+            }
+            *((int8_t *)value) = kw41zrf_get_ed_level(dev);
+            res = len;
+            break;
+
+        default:
+            break;
+    }
+
+    if (put_to_sleep_when_done) {
+        DEBUG("[kw41zrf] Go back to sleep\n");
+        kw41zrf_set_power_mode(dev, KW41ZRF_POWER_DSM);
+    }
+
+    if (res == -ENOTSUP) {
+        res = netdev_ieee802154_get((netdev_ieee802154_t *)netdev, opt, value, len);
+    }
+    return res;
+}
+
+static int kw41zrf_netdev_set(netdev_t *netdev, netopt_t opt, const void *value, size_t len)
+{
+    kw41zrf_t *dev = (kw41zrf_t *)netdev;
+    int res = -ENOTSUP;
+
+    if (dev == NULL) {
+        return -ENODEV;
+    }
+
+    /* These settings do not require the transceiver to be awake */
+    switch (opt) {
```

Comment:
Please add a case for `NETOPT_ADDRESS` and `NETOPT_ADDRESS_LONG` to the switch here returning the address directly from the device itself (so not from the netdev struct)